File: Java/LeetCode/K-diff Pairs in an Array.java
// https://leetcode.com/problems/k-diff-pairs-in-an-array/

class Solution {

    public int findPairs(int[] nums, int k) {
        Set<String> uniquePairs = new HashSet<>();

        for (int i = 0; i < nums.length; ++i) {
            for (int j = i+1; j < nums.length; ++j) {
                if (Math.abs(nums[i]-nums[j]) == k) {
                    int a = Math.min(nums[i], nums[j]);
                    int b = Math.max(nums[i], nums[j]);
                    uniquePairs.add(a + "," + b);
                }
            }
        }

        return uniquePairs.size();
    }
    
}

File: Kotlin/LeetCode/K-diff Pairs in an Array.kt
// https://leetcode.com/problems/k-diff-pairs-in-an-array/

class Solution {
  
    fun findPairs(nums: IntArray, k: Int): Int {
        val uniquePairs = mutableSetOf<String>();

        for (i in 0 until nums.size) {
            for (j in i+1 until nums.size) {
                if (abs(nums[i]-nums[j]) == k) {
                    val a = Math.min(nums[i], nums[j]);
                    val b = Math.max(nums[i], nums[j]);
                    uniquePairs.add("$a,$b");
                }
            }
        }

        return uniquePairs.size;
    }
    
}
